resources/functionMap.php

```diff
@@ -5166,7 +5166,7 @@
 'ImagickPixel::clear' => ['bool'],
 'ImagickPixel::clone' => ['void'],
 'ImagickPixel::destroy' => ['bool'],
-'ImagickPixel::getColor' => ['array', 'normalized='=>'bool'],
+'ImagickPixel::getColor' => ['array', 'normalized='=>'0|1|2'],
```

Comment:
you could also specify the return array shape:

```suggestion
'ImagickPixel::getColor' => ['array{r: int|float, g: int|float, b: int|float, a: int|float}', 'normalized='=>'0|1|2'],
```

this might be better off implemented as a return-type extension though, due to the nature of the 2nd arg and the way it affects the return types.

Reply:
Can you give me an example of such a return type extension? Then I can try to implement it.

Reply:
You can find a lot here https://github.com/phpstan/phpstan-src/tree/1.9.x/src/Type/Php

You'll need a DynamicMethodReturnTypeExtension, for instance https://github.com/phpstan/phpstan-src/blob/1.9.x/src/Type/Php/DateTimeModifyReturnTypeExtension.php
(You can see the whole PR here: https://github.com/phpstan/phpstan-src/pull/1267)

Reply:
and maybe we can write a conditional return type stub instead
https://phpstan.org/r/a8720857-2c4d-4bc9-a3bf-c8a46b51b02f

ex.
https://github.com/phpstan/phpstan-src/blob/2d4904895921ea3b37b5076c66690d4e067f9ed6/stubs/core.stub#L38-L42